### PR TITLE
translation(whd): add env_wrapper english file and correct mistake in its Chinese version 

### PR DESCRIPTION
**Pull request by walterwhd**

### CODE REVIEW

File: source/04_best_practice/env_wrapper.rst [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+- ObsNormEnv：The observed states are normalised according to the running mean and std.
+
+- RewardNormEnv： The environmental rewards are normalised according to the standard deviation of the runs (running mean and std).
+
+- RamWrapper： Converting the ram state of the original environment into an image-like state by extending the dimensionality of the observed state.
+
+- EpisodicLifeEnv：Let the death of an intelligence in the environment mark the end of an episode (game over), and only reset the game when the real game is over. In general, this helps the algorithm to estimate the value.
+
+- FireResetEnv：  Take ``fire`` action when the environment is reset. For more information <https://github.com/openai/baselines/issues/240>
```
Comment (TuTuHuss) [inline source/04_best_practice/env_wrapper.rst:38]: format problem :For more information <https://github.com/openai/baselines/issues/240>

**Comment by walterwhd** [comment]
@TuTuHuss

File: source/04_best_practice/env_wrapper.rst [hunk 2]
```diff
@@ -0,0 +1,118 @@
+Why do we need Env Wrapper
+------------------------------------------------------
+Environment module is one of the most vital modules in reinforcement learning。 We train our agents in these environmnets and we allow them to explore and learn in these envirnments。In addition to a number of benchmark environments （such as atari or mujoco）reinforcement learning may also include a variety of custom environments.Overall，The essence of the Env Wrapper is to add certain generic additional features to our custom environment.
```
Comment (kxzxvbk) [inline source/04_best_practice/env_wrapper.rst:3]: 。should be changed to .

Comment (kxzxvbk) [inline source/04_best_practice/env_wrapper.rst:3]: （ in chineese version should also be changed into (

File: source/04_best_practice/env_wrapper.rst [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+For instance：When we are training intelligences, we usually need to change the definition of the environment in order to achieve better training results, and these processing techniques are somewhat universal.For some environments, normalising the observed state is a very common pre-processing method. This processing makes training faster and more stable. If we extract this common part and put this preprocessing in an Env Wrapper, we avoid duplicate development. That is, if we want to change the way we normalise the observation state in the future, we can simply change it in this Env Wrapper.
```
Comment (kxzxvbk) [inline source/04_best_practice/env_wrapper.rst:4]: Maybe intelligence should be replaced with agent.

File: source/04_best_practice/env_wrapper.rst [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+Therefore, if the original environment is not perfectly adapted to our needs, we need to add functional modules to it to extend the functionality of the original environment and make it easy for the user to manipulate or adapt the environment's inputs and outputs. Env Wrapper is a simple solution for adding functionality.
+
+
+Env Wrapper offered by DI-engine
+==============================================
+
+DI-engine provides a large number of defined and generic Env Wrapper，The user can wrap it directly on top of the environment they need to use according to their needs.In the process of implementation，we refered  `OpenAI Baselines <https://github.com/openai/baselines/blob/master/baselines/common/atari_wrappers.py>`_ ，and folloiw the form of gym.Wrapper，which is `Gym.Wrapper <https://www.gymlibrary.dev/api/wrappers/>`_ ，In total, these include the following:
```
Comment (kxzxvbk) [inline source/04_best_practice/env_wrapper.rst:12]: defined -> pre-defined
，-> .
folloiw -> follow

File: source/04_best_practice/env_wrapper.rst [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+- NoopResetEnv：add a reset method to the environment. Resets the environment after some no-operations..
```
Comment (kxzxvbk) [inline source/04_best_practice/env_wrapper.rst:14]: resets -> reset

File: source/04_best_practice/env_wrapper.rst [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+
+- MaxAndSkipEnv： Each ``skip`` frame（doing the same action）returns the maximum value of the two most recent frames.(for max pooling across time steps)。
+
+- WarpFrame： Convert the size of the image frame to 84x84, such as `Nature‘s paper <https://www.deepmind.com/publications/human-level-control-through-deep-reinforcement-learning>`_  and it's following work。(Note that this registrar also converts RGB images to GREY images)
```
Comment (kxzxvbk) [inline source/04_best_practice/env_wrapper.rst:18]: such as -> according to
。 -> .

File: source/04_best_practice/env_wrapper.rst [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+
+- ScaledFloatFrame： Normalize status values to 0~1。
+
+- ClipRewardEnv： Cuts the reward to {+1, 0, -1} by the positive or negative of the reward.
```
Comment (kxzxvbk) [inline source/04_best_practice/env_wrapper.rst:22]: Cuts -> Clip

File: source/04_best_practice/env_wrapper.rst [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+
+- FrameStack： Set the nearest state frame of the stacked n_frames to the current state.
```
Comment (kxzxvbk) [inline source/04_best_practice/env_wrapper.rst:24]: Set the current state to be the stacked nearest n frames.

File: source/04_best_practice/env_wrapper.rst [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+
+- RunningMeanStd：a wrapper for updating variances, means and counts.
+
+- ObsNormEnv：The observed states are normalised according to the running mean and std.
+
+- RewardNormEnv： The environmental rewards are normalised according to the standard deviation of the runs (running mean and std).
+
+- RamWrapper： Converting the ram state of the original environment into an image-like state by extending the dimensionality of the observed state.
+
+- EpisodicLifeEnv：Let the death of an intelligence in the environment mark the end of an episode (game over), and only reset the game when the real game is over. In general, this helps the algorithm to estimate the value.
```
Comment (kxzxvbk) [inline source/04_best_practice/env_wrapper.rst:36]: intelligence -> agent

File: source/04_best_practice/env_wrapper.rst [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+
+- FireResetEnv：  Take ``fire`` action when the environment is reset. For more information please click `here <https://github.com/openai/baselines/issues/240>`_
+
+Tip: update_shape：This is a function that helps to identify the shape of observed states, actions and rewards after the env wrapper has been applied.
+
+How to use Env Wrapper
+------------------------------------
+The next question is how should we wrap the environment with Env Wrapper. One solution is to wrap the environment manually and explicitly：
```
Comment (kxzxvbk) [inline source/04_best_practice/env_wrapper.rst:44]: how should we -> how to

File: source/04_best_practice/env_wrapper.rst [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+                lambda env: ScaledFloatFrameWrapper(env)
+            ]
+        }
+    )
+
+
+In particular, the Env Wrappers in the list are wrapped outside the environment in order. In the example above, the Env Wrapper wraps a layer of MaxAndSkipWrapper and then a layer of ScaledFloatFrameWrapper, while the Env Wrapper serves to add functionality but does not change the original functionality.
+
+
+How to customise Env Wrapper （Example）
```
Comment (kxzxvbk) [inline source/04_best_practice/env_wrapper.rst:71]: How to customise an Env Wrapper

File: source/04_best_practice/env_wrapper.rst [hunk 12]
```diff
@@ -0,0 +1,118 @@
...
+-----------------------------------------
+Taking ObsNormEnv wrapper as an example。In order to normalis the observed state，we only need to change two methods in the original environment class：step method and reset method，The rest of the method remains the same.
```
Comment (kxzxvbk) [inline source/04_best_practice/env_wrapper.rst:73]: normalis -> normalise
。 -> .
， -> ,
： -> :

File: source/04_best_practice/env_wrapper.rst [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+Note that in some cases, as the normalised bounds of the observed state change, info will need to be modified accordingly.Please also note that the essence of the ObsNormEnv wrapper is to add additional functionality to the original environment, which is what the wrapper is all about. \
```
Comment (kxzxvbk) [inline source/04_best_practice/env_wrapper.rst:74]: info is needed to be modified accordingly

**Pull request closed without merging**